projects/pt1/python/torch_mlir_e2e_test/test_suite/reduction.py

```diff
@@ -1230,3 +1230,56 @@ def forward(self, input, target):
 @register_test_case(module_factory=lambda: CrossEntropyLossNoReductionModule())
 def CrossEntropyLossNoReductionModule_basic(module, tu: TestUtils):
     module.forward(tu.rand(8, 2), tu.randint(8, high=2))
+
+# ==============================================================================
+
+class TraceModule(torch.nn.Module):
+    def __init__(self) -> None:
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1], torch.float32, True),
+    ])
+    def forward(self, a):
+        return torch.ops.aten.trace(a)
+
+@register_test_case(module_factory=lambda: TraceModule())
+def TraceModule_basic(module, tu: TestUtils):
+    module.forward(tu.rand(3, 3))
+
+@register_test_case(module_factory=lambda: TraceModule())
+def TraceModule_nonsquare(module, tu: TestUtils):
+    module.forward(tu.rand(3, 4))
+
+@register_test_case(module_factory=lambda: TraceModule())
+def TraceModule_empty(module, tu: TestUtils):
+    module.forward(torch.empty(0,0))
+
+# ==============================================================================
+
+class TraceIntModule(torch.nn.Module):
+    def __init__(self) -> None:
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1], torch.int64, True),
+    ])
+    def forward(self, a):
+        return torch.ops.aten.trace(a)
+
+@register_test_case(module_factory=lambda: TraceIntModule())
```

Comment:
I see your logic accounts for rank-2 non-square tensors. Perhaps there should test for this case.

Reply:
Hi James, thanks for your review! Exactly, the logic accounts for rank-2 tensors. A test case for non-square and float-valued tensors is already contained, I could add one for int but do not think the handling of non-square tensors will depend on the dtype.

Reply:
So sorry, I missed the float tests above! My bad.

Reply:
No worries! :)